scrapi/harvesters/addis_ababa.py

```diff
@@ -6,6 +6,33 @@
 from __future__ import unicode_literals
 
 from scrapi.base import OAIHarvester
+from scrapi.base import helpers
+
+
+def oai_process_uris_addis_ababa(*args):
+
+    identifiers = helpers.gather_identifiers(args)
+    provider_uris, object_uris = helpers.seperate_provider_object_uris(identifiers)
+
+    potential_uris = (provider_uris + object_uris)
+
+    for i, uri in enumerate(potential_uris):
+        potential_uris[i].replace('http://handle.net/123456789/', 'http://etd.aa.edu.et/handle/123456789')
+
+    try:
+        canonical_uri = potential_uris[0]
+    except IndexError:
+        raise ValueError('No Canonical URI was returned for this record.')
+
+    provider_uris = potential_uris[0:len(provider_uris)]
```

Comment:
Does this have an off-by-one error? You might want `len(provider_uris) -1`.
Alternatively, you could alter the content of `provider_uris` and `object_uris` before concatenating them together, which would mean that you don't have to re-form these two lists at all.